RELEASING.md

```diff
@@ -1,11 +1,16 @@
 # Releasing
 
-- `Add steps to prepare release`
```

Comment:
Based on https://github.com/honeycombio/honeycomb-opentelemetry-android/pull/25, the only "prepare release" steps were to go through the codebase and manually update the version number.

However, now that we're taking it from the git tag directly, we don't need this at all, and can just skip straight to updating the Changelog + pushing a new tag.